### PR TITLE
fix(api): updated action service to use a queue to limit es query spam

### PR DESCRIPTION
**Pull request by cccs-mdr**
The existing (non-batched) action functionality wasn't scaling well with increased ingestion. This functionality should help with the growth of ingestion. Based heavily on the case rules functionality introduced in #327.

---

This pull request introduces a new background action queue system for batching and processing action executions, aiming to reduce Elasticsearch load during ingestion spikes. The main changes include adding a per-trigger Redis-backed action queue, a background worker for processing queued actions in batches, configuration options for the queue, and comprehensive integration tests to ensure correct behavior and batching. Several API endpoints and service methods have been updated to enqueue actions instead of executing them immediately, with a fallback to direct execution if the queue is disabled.

**Action Queue System Implementation**

* Added a new background worker (`api/howler/cronjobs/action_queue_worker.py`) that starts a daemon thread per trigger type, each draining its own Redis-backed queue and processing actions in batches with configurable size and timeout.
* Introduced the `ActionQueue` configuration in `config.py` to control enabling, batch size, and batch timeout for the action queue worker. [[1]](diffhunk://#diff-52ee7c226148cc8d84fc36b1a9b9dec6db434703478e48aca8a8fd2ce04b392fR356-R368) [[2]](diffhunk://#diff-52ee7c226148cc8d84fc36b1a9b9dec6db434703478e48aca8a8fd2ce04b392fR382-R383)

**Action Service Refactoring**

* Implemented `enqueue_action_execution` and supporting logic in `action_service.py` to buffer action execution requests to Redis queues, with a fallback to direct execution if the queue is disabled or unavailable. Also added `process_action_batch` to coalesce and execute batches efficiently.

**API and Service Usage Updates**

* Updated multiple API endpoints and service methods to use `enqueue_action_execution` instead of immediate execution, including hit creation, label addition/removal, and transitions. [[1]](diffhunk://#diff-c67f7d29b7497a7f1e9e3de768fd08a63c684ff518b4ba921228654e66f16111L126-R126) [[2]](diffhunk://#diff-c67f7d29b7497a7f1e9e3de768fd08a63c684ff518b4ba921228654e66f16111L554-R554) [[3]](diffhunk://#diff-c67f7d29b7497a7f1e9e3de768fd08a63c684ff518b4ba921228654e66f16111L612-R608) [[4]](diffhunk://#diff-31c99d34b517bb3a96b38f266ef680d00c1c8830375e2489f48416e4e0e1b32bL181-R181) [[5]](diffhunk://#diff-4329e287b29f436c78c60441e0b589847e05e0d428024ff7b3c2231c72553d95L741-R744)

**Testing Enhancements**

* Added and updated integration tests in `test_action_service.py` to cover the new queue-based execution, batching, duplicate coalescing, and edge cases, ensuring the queue worker processes actions as expected. [[1]](diffhunk://#diff-956a8525906864e99a1657449a8c02e97cad458d005929e188820f48395012f2R3-R35) [[2]](diffhunk://#diff-956a8525906864e99a1657449a8c02e97cad458d005929e188820f48395012f2L81-R120) [[3]](diffhunk://#diff-956a8525906864e99a1657449a8c02e97cad458d005929e188820f48395012f2L111-R129) [[4]](diffhunk://#diff-956a8525906864e99a1657449a8c02e97cad458d005929e188820f48395012f2L141-R341)

### CODE REVIEW

File: api/howler/services/action_service.py [hunk 1]
```diff
@@ -9,13 +9,98 @@
...
+        trigger: The trigger type (create, promote, demote, add_label, remove_label).
+        user: The user initiating the action.
+    """
+    if not hit_ids:
+        return
+
+    if not config.system.action_queue.enabled:
+        query = f"howler.id:({' OR '.join(sanitize_lucene_query(h) for h in hit_ids)})"
+        bulk_execute_on_query(query, trigger=trigger, user=user)
+        return
```
[the remark this answers is not in the document]
Comment (cccs-mdr) [reply api/howler/services/action_service.py:65]: This is down to a deployment issue.

File: api/howler/odm/models/config.py [hunk 2]
```diff
@@ -353,6 +353,19 @@ class ViewCleanup(BaseModel):
...
 
+class ActionQueue(BaseModel):
+    """Action queue worker configuration.
+
+    Controls the background worker that buffers action execution requests
+    via a Redis queue, coalescing them into batches to reduce Elasticsearch
+    query load during ingestion spikes.
+    """
+
+    enabled: bool = Field(default=True, description="Enable the action queue worker?")
```
[the remark this answers is not in the document]
Comment (cccs-mdr) [reply api/howler/odm/models/config.py:364]: Largely a deployment issue again.

File: api/howler/services/action_service.py [hunk 3]
```diff
@@ -9,13 +9,98 @@
...
+    """Return the action queue for *trigger*, creating it on first use."""
+    if trigger not in _action_queues:
+        _action_queues[trigger] = NamedQueue(
+            f"howler.action_queue.{trigger}",
+            host=config.core.redis.persistent.host,
+            port=config.core.redis.persistent.port,
+            private=False,
+        )
+
+    return _action_queues[trigger]
```
[the remark this answers is not in the document]
Comment (cccs-mb) [reply api/howler/services/action_service.py:42]: Its like buddy can't read the line literally the first line of this function

File: api/howler/cronjobs/action_queue_worker.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+logger = get_logger(__file__)
+
+_threads: dict[str, threading.Thread] = {}
+
+BATCH_SIZE: int = config.system.action_queue.batch_size
+BATCH_TIMEOUT: int = config.system.action_queue.batch_timeout
+
+
+def run_worker(trigger: str) -> None:  # pragma: no cover – long-running loop, tested via process_action_batch
```
Comment (cccs-mb) [inline api/howler/cronjobs/action_queue_worker.py:26]: Any value in disabling this if action_queue is disabled?

**Review by cccs-mb — APPROVED** [review]
LGTM! Just a question about the run_worker

File: api/howler/cronjobs/action_queue_worker.py (hunk 4 above)
Comment (cccs-mdr) [reply api/howler/cronjobs/action_queue_worker.py:26]: Done!

**Review by cccs-nicefield — APPROVED** [review]
